include/openthread/platform/radio.h

```diff
@@ -764,7 +778,8 @@ otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel);
  * Schedule a radio reception window at a specific time and duration.
  *
  * @param[in]  aChannel   The radio channel on which to receive.
- * @param[in]  aStart     The receive window start time, in microseconds.
+ * @param[in]  aStart     The receive window start time relative to the local
```

Comment:
@fgrandel  As discussed about the "window" definition, I wanted to make this more clear. Proposing the below clarification for the parameters:

```
/**
 * Schedule a radio reception window at a specific time and duration.
 *
 * @param[in]  aChannel   The radio channel on which to receive.
 * @param[in]  aStart     The receive window start time relative to the local radio clock, see `otPlatTimeGet`.
 *                        The window start time is when the radio is on and ready to receive the first symbol of 
 *                        the SHR.
 * @param[in]  aDuration  The receive window duration, in microseconds. The window end time is when the radio is 
 *                        turned off (in case it is not actively receiving a frame at that moment).
 *
 * @retval OT_ERROR_NONE    Successfully scheduled receive window.
 * @retval OT_ERROR_FAILED  The receive window could not be scheduled.
 */
otError otPlatRadioReceiveAt(otInstance *aInstance, uint8_t aChannel, uint32_t aStart, uint32_t aDuration);
```

Is this also what you had in mind? By using this time definition we avoid having to change various CSL code. An alternative here is to use the `mTimestamp` reference instead for `aStart`, but that's more impactful.

Reply:
Sorry, I may be fully wrong here. Although the original text nicely refers to the "window" concept (related to CSL sample) , if I look at the CSL code it seems to be assuming another form of timestamp reference. (e.g. it mentions "MHR" still.)

Was it the intention to use `otPlatRadioReceiveAt` with the exact same time reference (SFD based) as `mTimestamp` ? In that case the header could look like:

```
/**
 * Schedule a radio reception window at a specific time and duration.
 *
 * @param[in]  aChannel   The radio channel on which to receive.
 * @param[in]  aStart     The receive window start time relative to the local radio clock, see `otPlatTimeGet`.
 *                        The (SFD based) frame time reference is the same as for `mTimestamp`.
 * @param[in]  aDuration  The receive window duration, in microseconds. The time `aStart + aDuration` refers to the 
 *                        latest possible frame that can still be reliably detected within the window.
 *
 * @retval OT_ERROR_NONE    Successfully scheduled receive window.
 * @retval OT_ERROR_FAILED  The receive window could not be scheduled.
 */
otError otPlatRadioReceiveAt(otInstance *aInstance, uint8_t aChannel, uint32_t aStart, uint32_t aDuration);
```

Reply:
Or better, `aDuration` could simply refer to the time that the radio is on.

Reply:
> Is this also what you had in mind?

Yes, that's my understanding. @edmont Do you agree? I added you text just slightly re-formulated using SHALL/SHOULD terminology. See whether you like it.

Reply:
> Was it the intention to use `otPlatRadioReceiveAt` with the exact same time reference (SFD based) as `mTimestamp` ?

No, IMO neither MAC header nor SFD make for good RX window definitions on PHY/L1 level. Your initial comment and definition (the one that I also used now) represents exactly my understanding of a low-level RX window and I believe this is how the method is actually being implemented right now - at least in Zephyr.

Why not MAC header? "receive at" is below MAC level (L1/PHY). CSL channel samples are part of a MAC protocol (one of many IEEE 802.15.4 MAC sub-protocols). A generic "receive at" feature should not know anything about MAC level. It is true that CSL phase refers to start-of-MAC but that is very CSL specific and happens nowhere else in the standard. This definition should be converted to low level "receive at" in the MAC.

Why not SFD? Defining an RX window relative to SFD only works for very specific PHY configurations. IMO that would make the actual start of the RX window completely non-deterministic, as even within one PHY (e.g. SUN FSK) there may be several SHR lengths. And if your receiver supports several PHYs at the same time then you're completely lost.

IMO SFD is for RX/TX timestamps (including timed TX) - not more not less.

Reply:
> @edmont Do you agree?

I do. Thank you @fgrandel and @EskoDijk for taking the effort of making all these APIs parameters much more clear.